<a href="https://colab.research.google.com/github/shravani2022/Project/blob/main/electric_vehicles.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
import plotly.express as px
import plotly.graph_objects as go

In [ ]:
data = pd.read_csv('/content/ElectricCarData_Clean.csv')
data

Brand                            Model  AccelSec  TopSpeed_KmH  \
0         Tesla     Model 3 Long Range Dual Motor       4.6           233   
1    Volkswagen                         ID.3 Pure      10.0           160   
2      Polestar                                 2       4.7           210   
3           BMW                              iX3        6.8           180   
4         Honda                                e        9.5           145   
..           ...                              ...       ...           ...   
98       Nissan                       Ariya 63kWh       7.5           160   
99         Audi     e-tron S Sportback 55 quattro       4.5           210   
100      Nissan               Ariya e-4ORCE 63kWh       5.9           200   
101      Nissan   Ariya e-4ORCE 87kWh Performance       5.1           200   
102       Byton                 M-Byte 95 kWh 2WD       7.5           190   

     Range_Km  Efficiency_WhKm FastCharge_KmH RapidCharge PowerTrain  \
0         450              161            940         Yes        AWD   
1         270              167            250         Yes        RWD   
2         400              181            620         Yes        AWD   
3         360              206            560         Yes        RWD   
4         170              168            190         Yes        RWD   
..        ...              ...            ...         ...        ...   
98        330              191            440         Yes        FWD   
99        335              258            540         Yes        AWD   
100       325              194            440         Yes        AWD   
101       375              232            450         Yes        AWD   
102       400              238            480         Yes        AWD   

       PlugType  BodyStyle Segment  Seats  PriceEuro  
0    Type 2 CCS      Sedan       D      5      55480  
1    Type 2 CCS  Hatchback       C      5      30000  
2    Type 2 CCS   Liftback       D      5      56440  
3    Type 2 CCS        SUV       D      5      68040  
4    Type 2 CCS  Hatchback       B      4      32997  
..          ...        ...     ...    ...        ...  
98   Type 2 CCS  Hatchback       C      5      45000  
99   Type 2 CCS        SUV       E      5      96050  
100  Type 2 CCS  Hatchback       C      5      50000  
101  Type 2 CCS  Hatchback       C      5      65000  
102  Type 2 CCS        SUV       E      5      62000  

[103 rows x 14 columns]

In [ ]:
#pie chart of power train using plotly
power_count = data['PowerTrain'].value_counts().reset_index()
power_count

PowerTrain  count
0        AWD     41
1        FWD     37
2        RWD     25

In [ ]:
import plotly.express as px

# Assuming 'data' is your DataFrame and 'PowerTrain' is the column with power train types
power_train_counts = data['PowerTrain'].value_counts()
fig = px.pie(power_train_counts,
             values=power_train_counts.values,
             names=power_train_counts.index,
             title='Power Train Type Distribution')
fig.show()

# Analysis -
The pie chart shows the distribution of power train types, specifically between AWD, FWD, and RWD.
AWD: This category dominates the chart, representing approximately 39.98% . This indicates that AWD power trains are the most common among the data points.
FWD: This category comprises a significant portion  35.59% of the total. This suggests that FWD power trains are also minimum .
RWD: This category has the smallest share, representing 24.43% of the total. This indicates that RWD power trains are the least common among the data points.

In [ ]:
#bar of brand vs Euro
brand_euro = data.groupby('Brand')['PriceEuro'].mean().reset_index()
brand_euro
fig = px.bar(brand_euro, x='Brand', y='PriceEuro', title='Average Price (Euro) by Brand')
fig.show()

# Analysis -
The graph shows the average price (in Euros) of different brands. We can see that there's a wide range of prices across the brands.
High-Priced Brands: Some brands, such as Lightyear and Porsche, have significantly higher average prices compared to others.
Mid-Priced Brands: Many brands fall into the mid-price range, with average prices ranging from 40,000 to 80,000 Euros.
Low-Priced Brands: Some brands, like "SEAT", have the lowest average prices.



In [ ]:
px.scatter(data, x='Brand', y='FastCharge_KmH', color='PowerTrain', title='FastCharge_KmH by Brand', size='Seats')

In [ ]:
#line graph of brand vs topspeed_kmH
brand_topspeed = data.groupby('Brand')['TopSpeed_KmH'].mean().reset_index()
brand_topspeed
fig = px.line(brand_topspeed, x='Brand', y='TopSpeed_KmH', title='Average Top Speed by Brand',markers=True)
fig.show()

Analysis -
This graph shows the average top speed (in km/h) of different car brands. Each point represents a brand, and the height of the point along the vertical axis shows its top speed .Highest Top Speeds: Some brands, like Lucid, Porsche, and Tesla, reach much higher top speeds compared to others, going beyond 220 km/h. This indicates that these brands  focus on high-performance .Several brands such as Renault, Citroen, Fiat, and Smart have lower top speeds, mostly below 160 km/h.


In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Brand            103 non-null    object 
 1   Model            103 non-null    object 
 2   AccelSec         103 non-null    float64
 3   TopSpeed_KmH     103 non-null    int64  
 4   Range_Km         103 non-null    int64  
 5   Efficiency_WhKm  103 non-null    int64  
 6   FastCharge_KmH   103 non-null    object 
 7   RapidCharge      103 non-null    object 
 8   PowerTrain       103 non-null    object 
 9   PlugType         103 non-null    object 
 10  BodyStyle        103 non-null    object 
 11  Segment          103 non-null    object 
 12  Seats            103 non-null    int64  
 13  PriceEuro        103 non-null    int64  
dtypes: float64(1), int64(5), object(8)
memory usage: 11.4+ KB


In [ ]:
#pie chart of powertrain
power_train_counts = data['PowerTrain'].value_counts()
fig = px.pie(power_train_counts,
             values=power_train_counts.values,
             names=power_train_counts.index,
             title='Power Train Type Distribution',
             hole=0.5)
fig.show()



# Analysis -
This donut chart shows the distribution of different powertrain types (AWD, FWD, RWD) among vehicles.AWD (All-Wheel Drive): Makes up the largest portion at 39.8%, indicating it's the most common powertrain type in this dataset.
FWD (Front-Wheel Drive): Accounts for 35.9% of the vehicles. FWD is generally found in smaller .
RWD (Rear-Wheel Drive): Represents 24.3% of the vehicles.
from overall analysis we can see that AWD has highest percentages of vehicles


In [ ]:
px.scatter(data,x='Brand',y='TopSpeed_KmH',color='Segment',size='Seats',title='Top Speed by Brand and Segment')


# Analysis -
This scatter plot shows the top speeds of different car brands, with each point representing a car segment (indicated by color) and the size of each point representing the number of seats in the vehicle.
Some brands like Tesla and Lucid have models that reach higher top speeds, in the range of 250 km/h and above.
Other brands like Fiat, Renault, and Honda have models with lower top speeds, usually below 150 km/h.
The segments are represented by colors, with each segment generally clustering within certain speed ranges. For example, Segment D (blue) and Segment F (purple) often have higher top speeds, while Segment C (orange) tends to have moderate speeds.


In [ ]:
fig = go.Figure()
fig.add_trace(go.Bar(
    x=data["PowerTrain"],
    y=data["FastCharge_KmH"],
    name='FastCharge_KmH',
    marker_color='indianred'
))
fig.add_trace(go.Bar(
    x=data["PowerTrain"],
    y=data["TopSpeed_KmH"],
    name='TopSpeed_KmH',
    marker_color='lightsalmon'
))
fig.update_layout(barmode='group', xaxis_tickangle=-45)
fig.show()

# Analysis -
The graph shows the comparison of 2 variables: "FastCharge_kWh_50," and "Total_kWh_50" across three different time periods (years).
FastCharge_kWh: This variable shows a general upward trend, with the highest value in the last time period .

In [ ]:
#pie chart of segments
seg_count = data['Segment'].value_counts()

fig = px.pie(seg_count,
             values=seg_count.values,
             names=seg_count.index,
             labels=seg_count.index,
             title='Segment Distribution',
             hole=0.5)
fig.show()

# Analysis -
The pie chart shows the distribution of segments within the EV vehicle market.
Segment C: This segment dominates the chart, representing approximately 29.1% of the total. This indicates that Segment C is the most prevalent in the EV market.
Other Segments: The remaining segments (A, B, D, E, F, M, N, and S) have varying shares, ranging from 5.83% to 21.4%.